export symbol macro order matters for pointer types

```
3-24T16:06:40.1141752Z        "D:\a\ascent\ascent\build\conduit-v0.9.3\ALL_BUILD.vcxproj" (default target) (1) ->
2025-03-24T16:06:40.1142245Z        "D:\a\ascent\ascent\build\conduit-v0.9.3\libs\relay\conduit_relay.vcxproj" (default target) (9) ->
2025-03-24T16:06:40.1142620Z        (ClCompile target) -> 
2025-03-24T16:06:40.1143285Z          D:\a\ascent\ascent\source\conduit-v0.9.3\src\libs\relay\conduit_relay_io_silo_api.hpp(30,9): error C2059: syntax error: '__declspec(dllexport)' [D:\a\ascent\ascent\build\conduit-v0.9.3\libs\relay\conduit_relay.vcxproj]
2025-03-24T16:06:40.1144416Z          D:\a\ascent\ascent\source\conduit-v0.9.3\src\libs\relay\conduit_relay_io_silo_api.hpp(30,9): error C2059: syntax error: '__declspec(dllexport)' [D:\a\ascent\ascent\build\conduit-v0.9.3\libs\relay\conduit_relay.vcxproj]
2025-03-24T16:06:40.1145910Z          D:\a\ascent\ascent\source\conduit-v0.9.3\src\libs\relay\conduit_relay_io_silo_api.hpp(30,9): error C2059: syntax error: '__declspec(dllexport)' [D:\a\ascent\ascent\build\conduit-v0.9.3\libs\relay\conduit_relay.vcxproj]
2025-03-24T16:06:40.1147029Z          D:\a\ascent\ascent\source\conduit-v0.9.3\src\libs\relay\conduit_relay_io_silo_api.hpp(30,9): error C2059: syntax error: '__declspec(dllexport)' [D:\a\ascent\ascent\build\conduit-v0.9.3\libs\relay\conduit_relay.vcxproj]
2025-03-24T16:06:40.1148176Z          D:\a\ascent\ascent\source\conduit-v0.9.3\src\libs\relay\conduit_relay_io_silo.cpp(206,22): error C3861: 'silo_open_file_for_read': identifier not found [D:\a\ascent\ascent\build\conduit-v0.9.3\libs\relay\conduit_relay.vcxproj]

```

See:
https://stackoverflow.com/a/77662725

We need 

```
CONDUIT_RELAY_API DBfile*  silo_open_file_for_read(const std::string &path);
```

--- a/src/libs/relay/conduit_relay_io_silo_api.hpp
+++ b/src/libs/relay/conduit_relay_io_silo_api.hpp
@@ -27,7 +27,7 @@ bool CONDUIT_RELAY_API is_silo_file(const std::string &path,
 //-----------------------------------------------------------------------------
 /// Opens a silo file and returns a silo file handle
 //-----------------------------------------------------------------------------
-DBfile* CONDUIT_RELAY_API silo_open_file_for_read(const std::string &path);
+CONDUIT_RELAY_API DBfile* silo_open_file_for_read(const std::string &path);
 
 //-----------------------------------------------------------------------------
 /// Close silo file handle

--- a/src/libs/relay/conduit_relay_zfp.hpp
+++ b/src/libs/relay/conduit_relay_zfp.hpp
@@ -47,7 +47,7 @@ static const std::string ZFP_HEADER_SCALAR_TYPE_FIELD = "zfp_header_scalar_type"
 static const std::string ZFP_COMPRESSED_DATA_FIELD = "zfp_compressed_data";
 
 
-zfp::array* CONDUIT_RELAY_API unwrap_zfparray(const Node &node);
+CONDUIT_RELAY_API zfp::array* unwrap_zfparray(const Node &node);
 
 void CONDUIT_RELAY_API        wrap_zfparray(const zfp::array &arr,
                                             Node &node);

